src/prefect/settings.py

```diff
@@ -368,6 +368,36 @@ def check_for_deprecated_cloud_url(settings, value):
     return deprecated_value or value
 
 
+def check_for_misconfigured_api_url(api_url):
+    misconfigured_mappings = {
+        "app.prefect.cloud": (
+            "Warning: Your PREFECT_API_URL points to app.prefect.cloud. Did you mean"
+            " api.prefect.cloud?"
```

Comment:
I believe the `/api/` part is also a differentiating factor between the two URLs.
```suggestion
            " api.prefect.cloud/api?"
```

Reply:
I have another warning I'll push up in a separate commit but good call out